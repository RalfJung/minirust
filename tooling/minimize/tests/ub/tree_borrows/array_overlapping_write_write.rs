//@ compile-flags: --minimize-tree-borrows
fn main() {
    unsafe { 
        let mut data = [42u8, 57]; 
        let x = &mut data[0] as *mut u8; // (x, [R, R]) 
        let y = &mut data[1] as *mut u8; // (x, [R, R]) (y, [R, R]) 
        *x = 42; // (x, [A, R]) (y, [D, R]) 
        *y = 57; // (x, [A, D]) (y, [D, A]) 
    
        *x.add(1) = 42; // UB! Child Write to Disabled
    }
}
