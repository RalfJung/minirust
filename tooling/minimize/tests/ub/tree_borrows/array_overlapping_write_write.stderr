fatal error: UB: Tree Borrows: Child writing a pointer with Disabled permission
